### PR TITLE
Replace solve with solveset/linsolve in sympy.simplify

### PR DESCRIPTION
**Pull request by aktech**
TODO:
- [x] hyperexpand.py
- [x] ratsimp.py
- [x] simplify.py
- [x] test_simplify.py (Revert `solve` test)

Partial fix for #8711
@hargup  @flacjacket

### CODE REVIEW

File: sympy/simplify/ratsimp.py [hunk 1]
```diff
@@ -159,6 +159,10 @@ def _ratsimpmodprime(a, b, allsol, N=0, D=0):
                         order=opt.order, polys=True)[1]
 
             S = Poly(r, gens=opt.gens).coeffs()
+
+            # TODO: Replace solve with linsolve, as of now solving for
```
Comment (hargup) [inline sympy/simplify/ratsimp.py]: Please open an issue for this.

Comment (aktech) [inline sympy/simplify/ratsimp.py]: Currently, this is done by [`minsolve_linear_system`](https://github.com/sympy/sympy/blob/master/sympy/solvers/solvers.py#L1969) in `solvers.py` . I am not sure if  `linsolve` should implement this. Probably, the algorithm used here in `ratsimp.py` needs to be improved to use `linsolve`, as it doesn't looks clean to me.

**Comment by aktech** [comment]
Reverted removing solve tests.

**Comment by aktech** [comment]
The `solveset` API may change, so closing this as of now.

**Pull request closed without merging**